backend-api/src/functions/asyncIssueBiometricCredential/asyncIssueBiometricCredentialHandler.ts

```
@@ -7,26 +7,71 @@ import { logger } from "../common/logging/logger";
 import { LogMessage } from "../common/logging/LogMessage";
 import { setupLogger } from "../common/logging/setupLogger";
 import { validateVendorProcessingQueueSqsEvent } from "./validateSqsEvent";
+import { getIssueBiometricCredentialConfig } from "./issueBiometricCredentialConfig";
+import { Result, emptyFailure, successResult } from "../utils/result";
+import { GetSecrets } from "../common/config/secrets";
+
+export class RetainMessageOnQueue extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "RetainMessageOnQueue";
+  }
+}
 
 export async function lambdaHandlerConstructor(
-  _dependencies: IssueBiometricCredentialDependencies,
+  dependencies: IssueBiometricCredentialDependencies,
   event: SQSEvent,
   context: Context,
 ): Promise<void> {
   setupLogger(context);
   logger.info(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_STARTED);
 
+  const configResult = getIssueBiometricCredentialConfig(dependencies.env);
+  if (configResult.isError) {
+    logger.error(LogMessage.ISSUE_BIOMETRIC_CREDENTIAL_INVALID_CONFIG);
+    return;
```

question: Do we want to throw an error here so it can be retried? My only thinking is if there was an issue with the envVar would the canary then roll the lambda back and we'd want to have another go?

---

Tagging @sandymay-dd on this as it's an interesting point - should we retain messages on the queue for config errors (or perhaps other issues that suggest a likely config error like 4XXs on some network calls) because canaries will roll us back to a state where the issue may not be present and the message can be processed successfully?

---

Implemented throw new Error()